store-implementation/hbase-store/src/main/java/uk/gov/gchq/gaffer/hbasestore/operation/handler/AddElementsHandler.java

```diff
@@ -0,0 +1,143 @@
+import uk.gov.gchq.gaffer.store.Store;
+import uk.gov.gchq.gaffer.store.StoreException;
+import uk.gov.gchq.gaffer.store.operation.handler.OperationHandler;
+import uk.gov.gchq.gaffer.store.schema.SchemaElementDefinition;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * HBase will skip 'puts' if there are multiple 'puts' with the same rowId and column qualifier.
+ * To work around this issue, we need to aggregate elements within each batch before adding them to HBase.
+ * Due to this, optimising the batch size could have a big impact on performance.
+ * Configure the batch size using store property: hbase.writeBufferSize
+ */
+public class AddElementsHandler implements OperationHandler<AddElements> {
+    @Override
+    public Void doOperation(final AddElements operation,
+                            final Context context, final Store store)
+            throws OperationException {
+        addElements(operation, (HBaseStore) store);
+        return null;
+    }
+
+    private void addElements(final AddElements addElementsOperation, final HBaseStore store)
+            throws OperationException {
+        if (null == addElementsOperation.getInput()) {
+            return;
+        }
+
+        try {
+            final HTable table = store.getTable();
+            final boolean hasAggregators = store.getSchema().hasAggregators();
+            final String visibilityProperty = store.getSchema().getVisibilityProperty();
+            final Iterator<? extends Element> elements = addElementsOperation.getInput().iterator();
+            final ElementSerialisation serialisation = new ElementSerialisation(store.getSchema());
+            final int batchSize = store.getProperties().getWriteBufferSize();
+            final Map<Integer, Element> keyToElement = new HashMap<>(batchSize);
+            final Map<String, ElementAggregator> aggregators = new HashMap<>(store.getSchema().getEdges().size() + store.getSchema().getEntities().size());
+            while (elements.hasNext()) {
+                keyToElement.clear();
+                for (int i = 0; i < batchSize && elements.hasNext(); i++) {
+                    final Element element = elements.next();
+                    if (null == element) {
+                        i--;
+                        continue;
+                    }
+
+                    final int keyHashCode = new HashCodeBuilder(17, 37)
+                            .append(serialisation.getRowKeys(element).getFirst())
+                            .append(serialisation.getColumnVisibility(element))
+                            .append(serialisation.getColumnQualifier(element))
+                            .append(serialisation.getColumnVisibility(element))
+                            .toHashCode();
+                    final Element existingElement = keyToElement.get(keyHashCode);
```

Comment:
There's a possibility here that there will be collisions - 2 elements that hash to the same int. It's unlikely for small numbers of inserts, but not impossible. Ideally we'd avoid this.